homeassistant/components/tuya/fan.py

```diff
@@ -256,7 +256,7 @@ def percentage(self) -> int | None:
         if self._speeds is not None:
             if (value := self.device.status.get(self._speeds.dpcode)) is None:
                 return None
-            return ordered_list_item_to_percentage(self._speeds.range, value)
+            return ordered_list_item_to_percentage(self._speeds.range, str(value))
```

Comment:
This is not right. This is an enum type value. The fact it is communicating an numeric value, is a bug.

Please report this issue with Tuya, as they can fix this in their communications / API.

../Frenck